src/components/editMeasure/testCases/components/testCaseLanding/common/copyTestCases/CopyTestCaseDialog.tsx

```diff
@@ -251,240 +263,310 @@ const CopyTestCaseDialog = ({ open, onClose, onSubmit, measure }) => {
     setPage(0);
   };
 
+  const onSubmit = async (e) => {
+    if (_.isEmpty(selectedRowId)) {
+      return;
+    }
+    setExecuting(true);
+    setContinueDisabled(true);
+    testCaseServiceApi.current
+      .copyTestCasesToMeasure(
+        measure.id,
+        selectedRowId,
+        selectedTestCases?.map((tc: TestCase) => tc.id)
+      )
+      .then((copiedTestCaseIds: string[]) => {
+        if (
+          copiedTestCaseIds.length ===
+          selectedTestCases?.map((tc: TestCase) => tc.id).length
+        ) {
+          onClose("Test Cases have been successfully copied.", "success");
+        } else if (copiedTestCaseIds.length > 0) {
+          onClose(
+            "Test Cases have been successfully copied. Some Test Cases were invalid and were not copied.",
+            "warning"
+          );
+        } else {
+          onClose("Test Cases were invalid and not copied.", "danger");
+        }
+      })
+      .catch((e) => {
+        console.error(e);
+        onClose(
+          "Unable to copy Test Cases. Please contact the Help Desk.",
+          "danger"
+        );
+      })
+      .finally(() => {
+        setExecuting(false);
+      });
+  };
+
+  const handleDialogClose = async (e) => {
```

Comment:
does this need to be async?

Reply:
I don't know. I asked that same question and didn't find a suitable answer. Probably not?